src/engine/board.ts

```diff
@@ -7,8 +7,8 @@ export default class Board {
     public currentPlayer: Player;
     private readonly board: (Piece | undefined)[][];
 
-    public constructor() {
-        this.currentPlayer = Player.WHITE;
+    public constructor(currentPlayer?: Player) {
+        this.currentPlayer = currentPlayer ? currentPlayer : Player.WHITE;
```

Comment:
I don't think you wrote this but just FYI as it's useful this can be simplified to  `currentPlayer || Player.WHITE`
`X || Y ` means if X then X else Y, it is functionally the same just more concise